### ISSUE TITLE
Threshold of `false` seems confusing

### ISSUE THREAD

**Issue by elmigranto** [expressjs/compression#144]
Is `threshold` option being `false` means *never compress no matter response size*, or *default threshold of 1 KiB*, or something else?

``` js
// will compress any size
app.use(compression({threshold: 0}))

// returns JSON as plain text
app.use(compression({threshold: false}))
```

Makes sense to me that above two lines should be equivalent, but doesn't seem to be the case. Should either change this, or add more detailed description into README.

<hr />

❤️ to everyone working on this 😊

**Comment by elmigranto** [comment]
Judging by the following lines,  it's the *default threshold of 1 KiB*:

https://github.com/expressjs/compression/blob/7ebf291087d69156df4b086e2020e660a1dffb17/index.js#L55

So probably update to docs?

**Comment by elmigranto** [comment]
Here's how default is setup:

https://github.com/expressjs/compression/blob/7ebf291087d69156df4b086e2020e660a1dffb17/index.js#L55-L57

and here's a check:

https://github.com/expressjs/compression/blob/7ebf291087d69156df4b086e2020e660a1dffb17/index.js#L156-L160

So `{treshold: false}` results in never compressing, which is very surprising. Reasons for this:

 - `false == null` is `false`, so it's not 1024
 - `anyNumber < false` is only `true` when `anyNumber` is negative

**Comment by elmigranto** [comment]
Here's a README section I am talking about https://github.com/expressjs/compression/blob/7ebf291087d69156df4b086e2020e660a1dffb17/README.md#threshold.

**Comment by elmigranto** [comment]
I would proropse we disable length-checking for explicitly passed `false` value, but again, not sure what was intended behaviour here. Alternatively, we can throw on anything other than integer or valid byte strings removing the possibility to pass `false` entirely.

**Comment by elmigranto** [comment]
Again, to be clear, this is `debug` output I get for `{threshold: false}` on smallish response:

```
compression no compression: size below threshold +0ms
```

And here's for `{threshold: 0}`:

```
compression gzip compression +0ms
```

**Comment by elmigranto** [comment]
Oh, my bad, `{threshold: false}` compresses things greather than 1 KiB:

``` js
res.json({settings: app.settings, hugeString: '0'.repeat(1000)});
```

results in 

```
compression gzip compression +0ms

```

**Comment by dougwilson** [comment]
Hi @elmigranto sorry you were having trouble using the module. At first when I read your issue I was confused because I didn't even know that `false` was a value allowed for the `threshold` option at all. I looked at it was indeed listed in the docs as a value, though when I checked in the tests there is no tests for such a value.

The implementation clearly does not support the value, as you observed.

I took a look back at the history of this project, various points in time of the implementation show that `false` has never been a supported value and looking at how it got added in the readme, it looks like a mistake that it got there at all.

I'm going to fix the readme for the time being.

I never thought about `false` being a possible value, but I think that makes sense I suppose. If you are interested in seeing `false` be a supported value, you're welcome to make a pull request to support it, of course 👍
